src/test/java/com/booking/superride/controller/TaxiControllerTest.java

```diff
@@ -0,0 +1,57 @@
+package com.booking.superride.controller;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.*;
+
+import com.booking.superride.common.ContainersConfig;
+import com.booking.superride.domain.AddTaxiRequest;
+import io.restassured.http.ContentType;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+
+@Import(ContainersConfig.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class TaxiControllerTest {
+
+    @Test
+    void testSaveTaxiDetailsWithSingleTaxi() {
+        var request = new AddTaxiRequest("Tata", "TN 59 BH 8191", "Arul", 'A');
```

Comment:
Good way of writing input is using json.. if we use POJO and use `@JsonProperty` then we are not checking the correctness of input.